Extensions/postarchive.js

```diff
@@ -58,14 +65,20 @@ XKit.extensions.postarchive = {
 
 		}
 
-		$("#xkit_postarchive_inblog_button").click(function() {
-
-			var blog_url = $("#tumblelog_name").attr('data-tumblelog-name');
-
-			XKit.iframe.full();
+		window.addEventListener('message', function(e) {
+			if (e.origin !== blog_url && !e.data.xkit) { return; }
+			if (e.data.jquery_function) {
+				$('#xkit_postarchive_inblog_button')[e.data.jquery_function](e.data.args[0]);
```

Comment:
this is vaguely unsafe in that it could execute any arbitrary function—probably should just use a add_or_remove flag.